mobile/src/main/java/org/openhab/habdroid/ui/activity/ContentController.java

```diff
@@ -543,99 +554,115 @@ protected static int determineTransition(FragmentUpdateReason reason) {
+        protected static final String KEY_DRAWABLE = "drawable";
+        protected static final String KEY_BUTTON_1_TEXT = "button1text";
+        protected static final String KEY_BUTTON_1_TAG = "button1tag";
+        protected static final String KEY_BUTTON_2_TEXT = "button2text";
+        protected static final String KEY_BUTTON_2_TAG = "button2tag";
+        protected static final String KEY_PROGRESS = "progress";
+        protected static final int BUTTON_TAG_NONE = 0;
+        protected static final int BUTTON_TAG_OPEN_SETTINGS = 1;
+        protected static final int BUTTON_TAG_ENABLE_WIFI = 2;
+        protected static final int BUTTON_TAG_ENABLE_DEMO_MODE = 3;
+        protected static final int BUTTON_TAG_RETRY_NETWORK = 4;
+        protected static final int BUTTON_TAG_RETRY_SERVER_PROP_FETCH = 5;
+
+        protected static Bundle buildArgs(CharSequence message, @StringRes int buttonTextResId,
+                int buttonTag, @DrawableRes int drawableResId, boolean showProgress) {
+            return buildArgs(message, buttonTextResId, buttonTag, 0,
+                    0, drawableResId, showProgress);
+        }
+
+        protected static Bundle buildArgs(CharSequence message, @StringRes int button1TextResId,
+                int button1Tag, @StringRes int button2TextResId,
+                int button2Tag, @DrawableRes int drawableResId, boolean showProgress) {
             Bundle args = new Bundle();
-            args.putCharSequence("message", message);
-            args.putInt("drawable", drawableResId);
-            args.putInt("buttontext", buttonTextResId);
-            args.putBoolean("progress", showProgress);
+            args.putCharSequence(KEY_MESSAGE, message);
+            args.putInt(KEY_DRAWABLE, drawableResId);
+            args.putInt(KEY_BUTTON_1_TEXT, button1TextResId);
+            args.putInt(KEY_BUTTON_2_TEXT, button2TextResId);
+            args.putInt(KEY_BUTTON_1_TAG, button1Tag);
+            args.putInt(KEY_BUTTON_2_TAG, button2Tag);
+            args.putBoolean(KEY_PROGRESS, showProgress);
             return args;
         }
 
         @Override
-        public View onCreateView(@NonNull LayoutInflater inflater,
-                ViewGroup container, Bundle savedInstanceState) {
+        public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
+                Bundle savedInstanceState) {
             Bundle arguments = getArguments();
 
             View view = inflater.inflate(R.layout.fragment_status, container, false);
-
-            TextView descriptionText = view.findViewById(R.id.description);
-            CharSequence message = arguments.getCharSequence("message");
-            if (!TextUtils.isEmpty(message)) {
-                descriptionText.setText(message);
-            } else {
-                descriptionText.setVisibility(View.GONE);
-            }
+            setDescription(arguments, view, R.id.description, KEY_MESSAGE);
```

Comment:
Could restore original code here now that we have only one description.

Reply:
Will do